packages/@aws-cdk/aws-apigateway/lib/model.ts

```diff
@@ -1,5 +1,54 @@
+import cdk = require('@aws-cdk/cdk');
+import { CfnModel } from './apigateway.generated';
+import { IRestApi } from './restapi';
+
+export interface ModelImportProps {
+  readonly modelId: string;
+}
+
+export interface ModelProps {
+  /**
+   * The ID of a REST API with which to associate this model.
+   * Required.
+   */
+  restApi: IRestApi,
+
+  /**
+   * The content type for the model.
+   * Required.
+   */
+  contentType: string,
+
+  /**
+   * The schema to use to transform data to one or more output formats.
+   * Specify null ({}) if you don't want to specify a schema.
+   * Required.
+   * @see http://json-schema.org/
+   */
+  schema: any, // TODO: Validate that the schema provided matches the JSON schema spec.
```

Comment:
Yes please!

Reply:
Working on it, but JSII linting does not like the `$ref` and `$schema` properties in my JsonSchema interface. Attempting to remove those from the interface, but transform the output schema object to use $schema and $ref. It's getting kind of ugly with all the recursive properties...

Reply:
Alright, I've got a little mapper class to take care of transforming the `ref` and `schema` properties to include the $ prefix.